internal/common/voting/election.go

```diff
@@ -117,26 +120,202 @@ func (e *Election) bordaCountResult() shared.ClientID {
 
 	}
 
-	//Calculate the final score for all candidates and ditermine the winner.
+	//Calculate the final score for all candidates.
 	finalScore := make([]int, candidatesNumber)
 	for _, v := range preferenceMap {
 		for i := 0; i < candidatesNumber; i++ {
-			finalScore[i] = finalScore[i] + v[i]
+			finalScore[i] += v[i]
+		}
+	}
+	//variance is needed when two or more candidates have equal votes.
+	variance := make([]float64, candidatesNumber)
+	for _, v := range preferenceMap {
+		for i := 0; i < candidatesNumber; i++ {
+			vi := float64(v[i])
+			fSi := float64(finalScore[i])
+			cN := float64(candidatesNumber)
+			variance[i] += (vi - fSi/cN) * (vi - fSi/cN)
+		}
+	}
+
+	var totalScore float64 = 0
+	for i := 0; i < len(finalScore); i++ {
+		totalScore += float64(finalScore[i])
+	}
+
+	return finalScore, variance, totalScore
+}
+
+func findMaxScore(scoreList []int, variance []float64) (int, int) {
+	maxScore := 0
+	maxScoreIndex := 0
+	for i := 0; i < len(scoreList); i++ {
+		if scoreList[i] > maxScore {
+			maxScore = scoreList[i]
+			maxScoreIndex = i
+		}
+		if scoreList[i] == maxScore {
+			if variance[i] > variance[maxScoreIndex] {
+				maxScoreIndex = i
+			}
+		}
+	}
+	return maxScore, maxScoreIndex
+}
+
+func findMinScore(scoreList []int, variance []float64) (int, int) {
+	minScore := 0
+	for i := 0; i < len(scoreList); i++ {
+		minScore += scoreList[i]
+	}
+	minScoreIndex := 0
+	for i := 0; i < len(scoreList); i++ {
+		if scoreList[i] < minScore {
+			minScore = scoreList[i]
+			minScoreIndex = i
+		}
+		if scoreList[i] == minScore {
+			if variance[i] < variance[minScoreIndex] {
+				minScoreIndex = i
+			}
```

Comment:
See first comment

Reply:
Oh, thanks for telling me this, it is fixed now